packages/forms/src/UIForm/fieldsets/Tabs/Tabs.component.js

```diff
@@ -1,48 +1,70 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import { Tabs as RBTabs, Tab as RBTab } from 'react-bootstrap';
+import TabBar from '@talend/react-components/lib/TabBar';
 import classNames from 'classnames';
+import { translate } from 'react-i18next';
 
 import Fieldset from '../Fieldset';
 import { isValid } from '../../utils/validation';
 import theme from './Tabs.scss';
+import { I18N_DOMAIN_FORMS } from '../../../constants';
+import getDefaultT from '../../../translate';
 
-export default function Tabs(props) {
-	const { schema, ...restProps } = props;
-	const tabs = schema.items;
+class Tabs extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = { selectedKey: 0 };
+		this.onSelect = this.onSelect.bind(this);
+	}
 
-	return (
-		<RBTabs
-			className={classNames(theme['tf-tabs'], 'tf-tabs')}
-			id={`${restProps.id}-tabs`}
-		>
-			{tabs.map((tabSchema, index) => {
-				const tabIsValid = isValid(tabSchema, restProps.errors);
-				return (
-					<RBTab
-						eventKey={index}
-						key={index}
-						title={tabSchema.title}
-						tabClassName={classNames({ [theme['has-error']]: !tabIsValid })}
-					>
-						<Fieldset {...restProps} schema={tabSchema} />
-					</RBTab>
-				);
-			})}
-		</RBTabs>
-	);
+	onSelect(event, item) {
+		this.setState({ selectedKey: item.key });
+	}
+
+	render() {
+		const { schema, t, ...restProps } = this.props;
+		const tabs = schema.items.map((item, index) => {
+			const tabIsValid = isValid(item, restProps.errors);
```

Comment:
im puzzled, how a tab is supposed to be invalid ? What purpose does it serve ?

Reply:
It's something I added a while ago.
When you have an error in 1 tab, but don't display it's content, the submit will not be done, because of the error. But there is no indication because it is on a not-displayed input.
The idea is to give a visual indication (red tab) that there is an error in this tab.

And using only color is not good accessibility, so I added a text alternative.

Reply:
shouldn't it be handled by https://reactjs.org/docs/react-component.html#componentdidcatch

Reply:
not in this PR i mean

Reply:
By error I mean form error. If a field is mandatory for example, you press on submit, it will trigger a full validation, then add error messages on the fields.
But if it is not displayed because it's not in the active tab, you won't know that there is an error.
If the tab holds the info too, you will see it.